daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -692,40 +699,35 @@ class DataProcessor private (
 class ParseResult(dp: DataProcessor, override val resultState: PState)
   extends DFDL.ParseResult
   with WithDiagnosticsImpl
-  with ErrorHandler {
+  with Logging {
 
   /**
-   * To be successful here, we need to capture xerces parse/validation
+   * To be successful here, we need to capture parse/validation
    * errors and add them to the Diagnostics list in the PState.
    *
-   * @param state the initial parse state.
+   * @param bytes the parsed Infoset
    */
   def validateResult(bytes: Array[Byte]): Unit = {
     Assert.usage(resultState.processorStatus eq Success)
-    val schemaURIStrings = resultState.infoset.asInstanceOf[InfosetElement].runtimeData.schemaURIStringsForFullValidation
-    try {
-      val bis = new java.io.ByteArrayInputStream(bytes)
-      Validator.validateXMLSources(schemaURIStrings, bis, this)
-    } catch {
-      //
-      // Some SAX Parse errors are thrown even if you specify an error handler to the
-      // validator.
-      //
-      // So we also need this catch
-      //
-      case e: SAXException =>
-        resultState.validationErrorNoContext(e)
+
+
+    val v = dp.validationMode match {
+      case ValidationMode.Custom(cv) => cv
+      case _ =>
+        /// todo;; potential corner case (or not corner?) of needing this dynamic value to configure a validator
+        ///        resultState.infoset.asInstanceOf[InfosetElement].runtimeData.schemaURIStringsForFullValidation.mkString(",")
+        val v = ConfigValueFactory.fromAnyRef(resultState.infoset.asInstanceOf[InfosetElement].runtimeData.schemaURIStringsForFullValidation)
+        Validators.default.make(
+          Some(ConfigFactory.parseMap(Map(Validators.default.name() -> v).asJava))
```

Comment:
will add a function somewhere to make this a little more sightly.   Would be nice if these schema names could be obtained some other way at some other point, but I have found that way yet.

Reply:
Yeah, I've been unhappy with this InfosetElement vs. DIElement thing in Daffodil since shortly after putting it in. Not clear having that distinction adds any real value. The notion was that InfosetElement could be available on the API, and would expose only things API users need. But we end up downcasting them to DIElement all the time. 

But I don't think we need the downcast here. So I think you can shorten:
```
resultState.infoset.asInstanceOf[InfosetElement].... 
```
to drop the asInstanceOf. The PState.infoset is of type DIElement which isA InfosetElement already